.github/workflows/workflow.yml

```
@@ -181,11 +172,9 @@ jobs:
           echo "value=${main_size}" >> $GITHUB_OUTPUT
 
       - name: Use Xcode ${{ env.XCODE_VERSION }}
-        uses: maxim-lobanov/setup-xcode@v1
-        with:
-          xcode-version: ${{ env.XCODE_VERSION }}
+        run: sudo xcode-select --switch /Applications/Xcode_${XCODE_VERSION}.app
 
-      - uses: ruby/setup-ruby@v1
+      - uses: ruby/setup-ruby@922ebc4
```

```suggestion
      - uses: ruby/setup-ruby@922ebc4
```

please use full githash, rather than short.